### DISCUSSION TITLE
Unable to do non-collection to collection mapping

### DISCUSSION THREAD

**Discussion by cvsudheer108** [atlasmap/atlasmap#3356] [General]
Hi All, 
I have a requirement to do enterprise integration using Camel and there I need to map the messages from one format to another.

And added my messages to resources.
Here is my repo:
https://github.com/cvsudheer108/Atlasmap-Sample-Complex-Mapping

My requirement is to enrich a content as follows:
        from("timer:foo?period=5000")
            .setBody(simple("resource:classpath:postCodeRequest.json"))
            .log("--&gt; Sending: [${body}]")
                .to("direct:input-producer")
            .to("atlasmap:atlasmap-mapping.adm")
            .log("--&lt; Received: [${body}]");

        from("direct:input-producer")
                .setProperty("input-schema", simple("resource:classpath:inputRequest.json"))
                .log("-->; Input: [${exchangeProperty.input-schema}]");

Any my atlas map is like this:
<img width="1310" alt="Screenshot 2021-10-04 at 10 28 06 PM" src="https://user-images.githubusercontent.com/22774621/135892769-f74465e7-3d47-4473-bdad-925620b8e016.png">
Here I need to copy the source message to target schema and also enrich that with a few inout values.
The issue I am finding is when I need to map postcode to and target collection, that I need to identify like this:
groups/items[name="supplyPostcode"]/data to 2323
As can be seen in the mapping, the target can be chosen by an index only, but I will need to select by an xpath.

Tomohisa suggested to use f(x) on the field mapping, but I dont find it enabled for a non-collection to collection mapping.

Please suggest how can this mapping be resolved using AtlasMap.

Thank you!

**Comment by igarashitm** [comment]
That's not supported ATM. You'd need wait for https://github.com/atlasmap/atlasmap/issues/3357 or https://github.com/atlasmap/atlasmap/issues/1514

**Reply by cvsudheer108** [reply]
Thanks for the update @igarashitm
